### PR TITLE
Move snapshot validations to validation-webhook directory

### LINKED ISSUE #443: Move webhook logic from utils to validation-webhook folder
https://github.com/kubernetes-csi/external-snapshotter/pull/425/files#r538807152

### PR DESCRIPTION
**Pull request by prateekpandey14**
Signed-off-by: prateekpandey14 <prateekpandey14@gmail.com>

**What type of PR is this?**
 /kind cleanup

**What this PR does / why we need it**:
Refactor and move snapshot specific validations to `pkg/validation-webhook` folder

**Which issue(s) this PR fixes**:
Fixes #443 

**Does this PR introduce a user-facing change?**:

```release-note
NONE

```

### CODE REVIEW

File: pkg/validation-webhook/validation.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2020 The Kubernetes Authors.
```
Comment (xing-yang) [inline pkg/validation-webhook/validation.go:2]: 2021

Comment (prateekpandey14) [reply pkg/validation-webhook/validation.go]: done

**Pull request merged**